crates/context_aware_config/src/api/context/handlers.rs

```diff
@@ -864,35 +1025,51 @@ async fn weight_recompute(
                         condition: context.value.clone(),
                         old_weight: context.weight.clone(),
                         new_weight: val.clone(),
+                        description: context.description.clone(),
+                        change_reason: context.change_reason.clone(),
                     });
-                    Ok((val, context.id.clone()))
+                    Ok((
+                        val,
+                        context.id.clone(),
+                        context.description.clone(),
+                        context.change_reason.clone(),
+                    ))
                 }
                 Err(e) => {
                     log::error!("failed to calculate context weight: {}", e);
                     Err(unexpected_error!("Something went wrong"))
                 }
             }
         })
-        .collect::<superposition::Result<Vec<(BigDecimal, String)>>>()?;
+        .collect::<superposition::Result<Vec<(BigDecimal, String, String, String)>>>()?;
 
+    // Update database and add config version
     let last_modified_time = Utc::now().naive_utc();
     let config_version_id =
         conn.transaction::<_, superposition::AppError, _>(|transaction_conn| {
-            for (context_weight, context_id) in contexts_new_weight {
+            for (context_weight, context_id, _description, _change_reason) in contexts_new_weight.clone() {
                 diesel::update(contexts.filter(id.eq(context_id)))
-                    .set((weight.eq(context_weight), last_modified_at.eq(last_modified_time.clone()), last_modified_by.eq(user.get_email())))
-                    .execute(transaction_conn).map_err(|err| {
+                    .set((
+                        weight.eq(context_weight),
+                        last_modified_at.eq(last_modified_time.clone()),
+                        last_modified_by.eq(user.get_email()),
+                    ))
+                    .execute(transaction_conn)
+                    .map_err(|err| {
                         log::error!(
                             "Failed to execute query while recomputing weight, error: {err}"
                         );
                         db_error!(err)
                     })?;
             }
-            let version_id = add_config_version(&state, tags, transaction_conn)?;
+            let description = contexts_new_weight.iter().map(|(_, _, description, _)| description.clone()).collect::<Vec<String>>().join(",");
```

Comment:
instead of using change_reason and description of all contexts,
lets add only this 
description: weight recomputed
change_reason: weight recomputed